### ISSUE TITLE
Train M1 Selectional Preference model to filter implausible facts

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#687]
## Problem

Current system has no plausibility checking:
```
Fact: Fact(hund, ACTION, args=[object=telefon])
      "La hundo uzis la telefonon"

Is this plausible? Current system has NO IDEA.
Can't filter implausible facts from bad parses or hallucinations.
```

**Impact**: Accept garbage facts from parse errors

## Solution: Train M1 Selectional Preference Model

**Purpose**: Score plausibility of (subject, verb, object) triples

**Architecture**:
```
Input: (subj_root, verb_root, obj_root)
       ↓
Lookup embeddings (frozen from Stage 1)
       ↓
Concat: [subj_emb, verb_emb, obj_emb]
       ↓
MLP → plausibility_score (0-1)
```

## Implementation Steps

1. **Create training data export script** (Issue #621)
   ```python
   # Export S-V-O triples from v2.1 database
   python scripts/data/export_m1_triples_v2.1.py
   
   # Positive: Real triples from corpus
   #   (hund, vid, kat) → 1.0 (plausible)
   #   (persono, mang, pom) → 1.0 (plausible)
   
   # Negative: Synthetic (shuffle corpus triples)
   #   (hund, vid, hieraŭ) → 0.0 (implausible - not entity)
   #   (tablo, pens, libr) → 0.0 (implausible - tables don't think)
   
   # Size: ~10M triples
   ```

2. **Implement M1 model**
   - MLP over concatenated root embeddings
   - Binary classification (plausible/implausible)
   - Freeze root embeddings from Stage 1

3. **Integrate into fact extraction pipeline**
   - Score all extracted facts
   - Filter facts below threshold (e.g., 0.3)
   - Log filtered facts for debugging

## Expected Impact

- Fact filtering: Remove ~20-30% of garbage facts
- Answer quality: Cleaner, more coherent
- Implementation time: 2 weeks

## Dependencies

- #617, #618 (root embeddings training - must freeze these)
- #621 (M1 training data export script)

## Testing

- Accuracy on held-out test set
- Precision/recall on synthetic implausible triples
- End-to-end answer quality improvement

## Related

See `/tmp/rag_improvement_roadmap.md` - Priority 3

**Comment by marctjones** [comment]
## Update: Being Replaced by Semantic Type Hierarchy Approach

The original M1 Selectional Preference model (10M params) is being replaced by a more efficient **Semantic Fact Validator** (500K params) using automated semantic type discovery.

**New Approach** - See #699:
- ✅ 20x smaller (500K vs 10M params)
- ✅ Fully automated from corpus (zero human annotation)
- ✅ Interpretable via semantic type labels
- ✅ Two-layer hybrid (deterministic + statistical)

**Pipeline**:
1. Extract SVO triples from corpus (4.5-5.5M triples)
2. Cluster roots by verb co-occurrence → 18 semantic types
3. Generate verb selectional preferences P(type|verb)
4. Train 500K param validator

**Status**: Scripts ready (#699), waiting to run pipeline on full corpus

This issue can be closed once #699 is complete and the new SFV is integrated.

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
